### PR TITLE
Deps.dev system check

### PR DESCRIPTION
**Pull request by pxp928**
Fix for now for https://github.com/guacsec/guac/issues/749 where the maven does not use "/" but ":" as a separator between the namespace and name.

Add more unit tests to cover each type of system supported by deps.dev.

~Depends on #746~

### CODE REVIEW

File: pkg/ingestor/parser/deps_dev/deps_dev.go [hunk 1]
```diff
@@ -182,5 +138,16 @@ func (d *depsDevParser) GetIdentities(ctx context.Context) []common.TrustInforma
 }
 
 func (d *depsDevParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
-	return nil, fmt.Errorf("not yet implemented")
+	idstrings := &common.IdentifierStrings{}
```
Comment (pxp928) [inline pkg/ingestor/parser/deps_dev/deps_dev.go:141]: Depends on #746

File: pkg/handler/collector/deps_dev/deps_dev.go [hunk 2]
```diff
@@ -302,11 +308,20 @@ func (d *depsCollector) collectAdditionalMetadata(ctx context.Context, pkgType s
 
 func getVersionKey(pkgType string, namespace *string, name string, version *string) (*pb.VersionKey, error) {
 	queryName := ""
-	if namespace != nil && *namespace != "" {
-		queryName = strings.TrimSuffix(*namespace, "/") + "/" + name
+	if pkgType != "maven" {
```
Comment (lumjjb) [inline pkg/handler/collector/deps_dev/deps_dev.go:311]: is it worth creating an issue with label "data-quality" to talk about this? or is this specific to deps.dev?

**Review by lumjjb — APPROVED** [review]

File: pkg/handler/collector/deps_dev/deps_dev.go (hunk 2 above)
Comment (pxp928) [reply pkg/handler/collector/deps_dev/deps_dev.go:311]: This is specific to deps.dev

File: pkg/ingestor/parser/deps_dev/deps_dev.go (hunk 1 above)
Comment (pxp928) [reply pkg/ingestor/parser/deps_dev/deps_dev.go]: rebased!

**Pull request merged**